lib/rails/dom/testing/assertions/dom_assertions.rb

```diff
@@ -7,43 +7,61 @@ module DomAssertions
+          def assert_dom_equal(expected, actual, message = nil, strict: false)
             expected_dom, actual_dom = fragment(expected), fragment(actual)
             message ||= "Expected: #{expected}\nActual: #{actual}"
-            assert compare_doms(expected_dom, actual_dom), message
+            assert compare_doms(expected_dom, actual_dom, strict), message
           end
 
           # The negated form of +assert_dom_equal+.
           #
           #   # assert that the referenced method does not generate the specified HTML string
           #   assert_dom_not_equal '<a href="http://www.example.com">Apples</a>', link_to("Oranges", "http://www.example.com")
-          def assert_dom_not_equal(expected, actual, message = nil)
+          def assert_dom_not_equal(expected, actual, message = nil, strict: false)
             expected_dom, actual_dom = fragment(expected), fragment(actual)
             message ||= "Expected: #{expected}\nActual: #{actual}"
-            assert_not compare_doms(expected_dom, actual_dom), message
+            assert_not compare_doms(expected_dom, actual_dom, strict), message
           end
 
           protected
 
-            def compare_doms(expected, actual)
-              return false unless expected.children.size == actual.children.size
+            def compare_doms(expected, actual, strict)
+              expected_children = extract_children(expected, strict)
+              actual_children   = extract_children(actual, strict)
+              return false unless expected_children.size == actual_children.size
 
-              expected.children.each_with_index do |child, i|
-                return false unless equal_children?(child, actual.children[i])
+              expected_children.each_with_index do |child, i|
+                return false unless equal_children?(child, actual_children[i], strict)
               end
 
               true
             end
 
-            def equal_children?(child, other_child)
+            def extract_children(node, strict)
+              if strict
+                node.children
+              else
+                node.children.reject{|n| n.text? && n.text.blank?}
+              end
+            end
+
+            def equal_children?(child, other_child, strict)
               return false unless child.type == other_child.type
 
               if child.element?
                 child.name == other_child.name &&
                     equal_attribute_nodes?(child.attribute_nodes, other_child.attribute_nodes) &&
-                    compare_doms(child, other_child)
+                    compare_doms(child, other_child, strict)
               else
+                equal_child?(child, other_child, strict)
+              end
+            end
+
+            def equal_child?(child, other_child, strict)
+              if strict
                 child.to_s == other_child.to_s
+              else
+                child.to_s.strip == other_child.to_s.strip
```

Comment:
Instead of / in addition to `strip`, I think we need some kind of regex that collapses interior white spaces as well. For example,

```html
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p><p>Curabitur pretium tincidunt lacus. Nulla gravida orci a odio. Nullam varius, turpis et commodo pharetra, est eros bibendum elit, nec luctus magna felis sollicitudin mauris. Integer in mauris eu nibh euismod gravida. Duis ac tellus et risus vulputate vehicula. Donec lobortis risus a elit. Etiam tempor. Ut ullamcorper, ligula eu tempor congue, eros est euismod turpis, id tincidunt sapien risus a quam. Maecenas fermentum consequat mi. Donec fermentum. Pellentesque malesuada nulla a mi. Duis sapien sem, aliquet nec, commodo eget, consequat quis, neque. Aliquam faucibus, elit ut dictum aliquet, felis nisl adipiscing sapien, sed malesuada diam lacus eget erat. Cras mollis scelerisque nunc. Nullam arcu. Aliquam consequat. Curabitur augue lorem, dapibus quis, laoreet et, pretium ac, nisi. Aenean magna nisl, mollis quis, molestie eu, feugiat in, orci. In hac habitasse platea dictumst.</p>
```

vs 


```html
<p>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
  tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
  quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
  consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
  cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
  proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</p>

<p>
  Curabitur pretium tincidunt lacus. Nulla gravida orci a odio. Nullam varius,
  turpis et commodo pharetra, est eros bibendum elit, nec luctus magna felis
  sollicitudin mauris. Integer in mauris eu nibh euismod gravida. Duis ac
  tellus et risus vulputate vehicula. Donec lobortis risus a elit. Etiam
  tempor. Ut ullamcorper, ligula eu tempor congue, eros est euismod turpis, id
  tincidunt sapien risus a quam. Maecenas fermentum consequat mi. Donec
  fermentum. Pellentesque malesuada nulla a mi. Duis sapien sem, aliquet nec,
  commodo eget, consequat quis, neque. Aliquam faucibus, elit ut dictum
  aliquet, felis nisl adipiscing sapien, sed malesuada diam lacus eget erat.
  Cras mollis scelerisque nunc. Nullam arcu. Aliquam consequat. Curabitur
  augue lorem, dapibus quis, laoreet et, pretium ac, nisi. Aenean magna nisl,
  mollis quis, molestie eu, feugiat in, orci. In hac habitasse platea dictumst.
</p>
```

Reply:
I added a test case for this [here](https://github.com/rails/rails-dom-testing/pull/84/files#diff-c12bb7844bbbfd4d90bc5fbc0d738344R103) and solved it by using `split` instead of `strip` on both strings.